app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/WhatIsYourOrganisationsAddressController.scala

```diff
@@ -54,60 +56,73 @@ class WhatIsYourOrganisationsAddressController @Inject() (
-      val filledForm = contactAddressForm.bindFromRequest()
+      subscriptionDetailsService.cachedOrganisationType.flatMap { optOrgType =>
+        val orgType = optOrgType.getOrElse(throw DataUnavailableException("organisation type unavailable"))
 
-      val (countriesToInclude, countriesInCountryPicker) =
-        Countries.getCountryParameters(requestSessionData.selectedUserLocationWithIslands)
+        val filledForm = contactAddressForm.bindFromRequest()
 
-      if (filledForm.hasErrors)
-        Future.successful(
-          BadRequest(
-            what_is_your_organisations_address_view(
-              isInReviewMode,
-              filledForm,
-              countriesToInclude,
-              countriesInCountryPicker,
-              EmbassyId,
-              service
+        val (countriesToInclude, countriesInCountryPicker) =
+          Countries.getCountryParameters(requestSessionData.selectedUserLocationWithIslands)
+
+        if (filledForm.hasErrors)
+          Future.successful(
+            BadRequest(
+              what_is_your_organisations_address_view(
+                isInReviewMode,
+                filledForm,
+                countriesToInclude,
+                countriesInCountryPicker,
+                orgType.id,
+                service
+              )
             )
           )
-        )
-      else {
-        val addr = filledForm.value.head
-        registrationDetailsService.cacheAddress(
-          Address(addr.lineOne, addr.lineTwo, Some(addr.townCity), None, Some(addr.postcode), addr.country)
-        ).flatMap { _ =>
-          subscriptionDetailsService.cacheAddressDetails(filledForm.value.head)
-            .flatMap(_ => subscriptionFlowManager.startSubscriptionFlowWithPage(None, service, redirectFlowLocation))
-            .map {
-              case (flowPageOne, session) => Redirect(flowPageOne.url(service)).withSession(session)
-            }
+        else {
+          val addr = filledForm.value.head
+          registrationDetailsService.cacheAddress(
+            Address(addr.lineOne, addr.lineTwo, Some(addr.townCity), None, Some(addr.postcode), addr.country)
+          ).flatMap { _ =>
+            subscriptionDetailsService.cacheAddressDetails(filledForm.value.head)
+              .flatMap(
+                _ => subscriptionFlowManager.startSubscriptionFlowWithPage(None, service, redirectFlowLocation(orgType))
+              )
+              .map {
+                case (flowPageOne, session) => Redirect(flowPageOne.url(service)).withSession(session)
+              }
 
+          }
         }
       }
     }
   }
 
-  private def redirectFlowLocation(implicit request: Request[AnyContent]): SubscriptionFlow = {
+  private def redirectFlowLocation(
+    cdsOrganisationType: CdsOrganisationType
+  )(implicit request: Request[AnyContent]): SubscriptionFlow = {
     if (requestSessionData.selectedUserLocation.contains(UserLocation.Iom)) {
```

Comment:
Personally, my brain hurts when reading nested if-else statements. I believe the logic can be simplified as:
```
if (requestSessionData.selectedUserLocation.contains(UserLocation.Iom) && cdsOrganisationType == Partnership)
  PartnershipSubscriptionFlowIom
else 
  CharityPublicBodySubscriptionNoUtrFlowIom
```
Or this one, which I prefer more:
```
(requestSessionData.selectedUserLocation, cdsOrganisationType) match {
  case (Some(UserLocation.Iom), Partnership) => PartnershipSubscriptionFlowIom
  case _                                     => CharityPublicBodySubscriptionNoUtrFlow
}
```